src/Nethermind/Nethermind.Consensus/Validators/HeaderValidator.cs

```diff
@@ -236,7 +236,8 @@ protected virtual bool ValidateTotalDifficulty(BlockHeader parent, BlockHeader h
             if (header.TotalDifficulty == 0)
             {
                 // Same as in BlockTree.SetTotalDifficulty
-                if (!(_blockTree.Genesis!.Difficulty == 0 && _specProvider.TerminalTotalDifficulty == 0))
+                // TODO: Optimism validate td
+                if (!(_blockTree.Genesis!.Difficulty == 0 && _specProvider.TerminalTotalDifficulty == 0) && (_specProvider.ChainId != 420 && _specProvider.ChainId != 10))
```

Comment:
Care to elaborate this check? There are several unknown constants like `420` and `10`.